src/workspaces/dashboard/WorkspaceRightBoxSection.ts

```diff
@@ -0,0 +1,38 @@
+import { useCallback } from 'react';
+import { h } from 'react-hyperscript-helpers';
+import { RightBoxSection, RightBoxSectionProps } from 'src/components/RightBoxSection';
+import { Ajax } from 'src/libs/ajax';
+import Events, { extractWorkspaceDetails } from 'src/libs/events';
+import { WorkspaceWrapper } from 'src/workspaces/utils';
+
+interface WorkspaceRightBoxSectionProps extends RightBoxSectionProps {
+  workspace: WorkspaceWrapper; // used for metrics eventing
+}
+
+export const WorkspaceRightBoxSection = (props: WorkspaceRightBoxSectionProps) => {
+  const { workspace, title, persistenceId, defaultPanelOpen, afterTitle, info, children } = props;
+
+  const workspaceDashboardToggle = useCallback(
+    (panelOpen: boolean) => {
+      Ajax().Metrics.captureEvent(Events.workspaceDashboardToggleSection, {
+        title,
+        opened: panelOpen,
+        ...extractWorkspaceDetails(workspace),
+      });
+    },
+    [title, workspace]
+  );
+
+  return h(
```

Comment:
We are in the process of converting from hyperscript helpers to TSX; we certainly don't want to be adding new files that use hyperscript helpers.

Could you please convert this to TSX? It should be a straightforward replacement (GitHub copilot can help you!).